infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/KubernetesTrustedCAProvisioner.java

```diff
@@ -114,7 +114,9 @@ public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity runtimeID)
 
     KubernetesNamespace namespace = namespaceFactory.getOrCreate(runtimeID);
     Optional<ConfigMap> existing = namespace.configMaps().get(configMapName);
-    if (existing.isEmpty() || !existing.get().getData().equals(allCaCertsConfigMap.getData())) {
+    if (existing.isEmpty()
+        || !(existing.get().getData() == allCaCertsConfigMap.getData()
```

Comment:
`A || !(B || C) ` is hard to read and understand.
The following can be a bit easier if `A || !B && !C!`
If empty OR not the same (not equals in pointers terms && does not have the same values)

WDYT?

Reply:
Conditions B and C are logically related so, to my mind, it is clearer to leave that as is.